schema-engine/core/src/state.rs

```diff
@@ -371,9 +376,9 @@ impl GenericApi for EngineState {
                         });
 
                         Ok(IntrospectResult {
-                            datamodel: result.data_model,
+                            datamodel: datamodels.remove(0).1,
```

Comment:
This is how we, for now, make multi-file introspection back to a single file at the surface API. This is fine because introspection cannot, atm, take more than one input file and is thus guaranteed to return at most one file.